pkg/apis/core/validation/validation.go

```diff
@@ -4050,6 +4086,13 @@ func validateSeccompAnnotationsAndFieldsMatch(annotationValue string, seccompFie
 // ValidatePodUpdate tests to see if the update is legal for an end user to make. newPod is updated with fields
 // that cannot be changed.
 func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) field.ErrorList {
+	updatableSpecFields := []string{
```

Comment:
define this outside the func, then make a local copy when needed (or just define 2 and use the gate to decide which to use) ?

Reply:
I see this pattern with validVolumeDownwardAPIFieldPathExpressions. Just for my education, what's reason do to this? If we end up making a copy, we still use up stack space, i think.

Reply:
Previous mistakes do not justify future mistakes :)

There's really no reason for it to be local and dynamic.  I would just define 2 non-local variables and when the gate eventually disappears, so does the 2nd definition.

Reply:
fixed in commit https://github.com/kubernetes/kubernetes/pull/102884/commits/b60c5d045bb85eacae064e209994c4afe22a4e5f